### PR TITLE
add message signer

### PR DESCRIPTION
**Pull request by abenso**
Implemented a new feature for signing custom messages

### CODE REVIEW

File: src/app.ts [hunk 1]
```diff
@@ -19,17 +19,18 @@ import { ByteStream } from '@zondax/ledger-js/dist/byteStream'
...
 import { Account, AccountType, ResponseAddress, VaultAccount } from './types'
-import { ResponseSign } from './types'
+import { MsgSigner, ResponseSign } from './types'
 
 export class SpaceMeshApp extends BaseApp {
     static _INS = {
         GET_VERSION: 0x00 as number,
         GET_ADDR: 0x01 as number,
         SIGN: 0x02 as number,
+        SIGN_MESSAGE: 0x03 as number,
```
Comment (emmanuelm41) [inline src/app.ts:29]: This won't be backward compatible, as we are changing meaning to current values. Isn't that a problem?

File: src/app.ts [hunk 2]
```diff
@@ -153,4 +154,30 @@ export class SpaceMeshApp extends BaseApp {
...
+        bs.appendUint16(signMessage.message.length)
+        bs.appendBytes(signMessage.prefix)
+        bs.appendUint8(signMessage.domain)
+        bs.appendBytes(signMessage.message)
+
+        const payload = bs.getCompleteBuffer()
+
+        const chunks = this.prepareChunks(path, payload)
+        try {
+            let signatureResponse = await this.signSendChunk(this.INS.SIGN_MESSAGE, 1, chunks.length, chunks[0])
```
Comment (emmanuelm41) [inline src/app.ts:170]: Is this one starting from 1? I am not 100% into the signSendChunk function.

**Review by emmanuelm41 — COMMENTED** [review]
Just a few comments to address

File: src/app.ts (hunk 2 above)
Comment (abenso) [reply src/app.ts:170]: Yes, this have to start in 1

File: src/app.ts (hunk 1 above)
Comment (abenso) [reply src/app.ts]: I fixed to keep backward compatible

**Review by emmanuelm41 — APPROVED** [review]
Nice! We will need to change the class name in the future, as SpacemeshApp should be the correct name (as far as I remember). Could be done on another PR

**Pull request merged**